pkg/repositories/repositories.go

```diff
@@ -58,6 +58,41 @@ func NewConfig(filename string) (Config, error) {
 	return repoConfig, nil
 }
 
+// UpdateConfigVersions updates a config in-place, setting the afterVersion to
+// its version, and it's version to a newer one specified in a secondary config
+func UpdateConfigVersions(baseConfig *Config, configWithNewVersions *Config) {
+	// Extract new repos regardless of category
+	var newRepos []Repository
```

Comment:
This variable is being stored solely for use by `checkForNewVersion` below. Every invocation of `checkForNewVersion` loops through newRepos which is duplicating work. Perhaps we can overcome this by making this a map on the very value that identifies repos i.e. URL. Then this becomes:
```
newRepos := map[string]Repository{}
```

AND `checkForNewVersion` can be got rid of with:

```
newRepo, present := newRepos[currentRepo.URL]
``` 

The condition for existence of a newRepo is now captured by the variable `present` and then we can have
```
if !present {
   continue
}

// set remappedRepo
remappedRepo := currentRepo
remappedRepo.AfterVersion = currentRepo.Version
remappedRepo.Version = newRepo.Version
```

You'll notice that I've inverted the conditional so that we're now using line of right. The happy path is the one that we follow down. See https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88